### ISSUE TITLE
Javascript/ES6: Tagged Template Literals support

### ISSUE THREAD

**Issue by kgnugur** [andymass/vim-matchup#58]
> `

   should work the same way `(` In the second line

```javascript
// These are equivalent:
fn`some string here`
fn(['some string here'])
```

[MDN Doc on Tagged Template Literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates)

**Comment by andymass** [comment]
Does the `fn` part matter at all?  Are you suggesting that `%` should move between the two backticks?

**Comment by kgnugur** [comment]
Yes

**Comment by davidsierradz** [comment]
Is this working? I have in my `after/ftplugin/javascript.vim`:

```viml
set matchpairs+=`:`
"set matchpairs? -> matchpairs=(:),{:},[:],<:>,`:`
```
inside a javascript file:

```javascript
`foo`

`bar`
```

The first foo backtick doesnt work, the second foo backtick goes to the first foo backtick, the first bar backtick goes to the second foo backtick and the second bar backtick goes to the first bar backtick:

```
+---+
|   |
v   |
`foo`
    ^
    |
+---+
|
`bar`
^   |
|   |
+---+
```

Thanks

**Comment by andymass** [comment]
No, it's not currently supported.  Unfortunately, it's difficult to accomplish, since there's no way to tell what is the start and what is the end.  I think it would either need to rely on syntax highlighting, or count backticks on a single line.

Can tagged template literals span lines?

**Comment by davidsierradz** [comment]
> Can tagged template literals span lines?

Yeah

I think if the implementation is difficult and/or performance is impacted, then this is not worth it (a nice to have at least)

**Comment by ashnur** [comment]
> No, it's not currently supported. Unfortunately, it's difficult to accomplish, since there's no way to tell what is the start and what is the end. 

Why? If the index of `` ` `` is even, it's the start, if  it's odd it's the end. Only thing important is to not confuse ```` ``` ```` with ``` `` ``` or `` ` ``
